core/circuit_breaker/circuit_breaker.go

```diff
@@ -2,252 +2,521 @@ package circuit_breaker
 
 import (
 	"sync/atomic"
-	"time"
+	"unsafe"
 
 	"github.com/alibaba/sentinel-golang/core/base"
-	"github.com/alibaba/sentinel-golang/core/stat"
+	sbase "github.com/alibaba/sentinel-golang/core/stat/base"
 	"github.com/alibaba/sentinel-golang/util"
 )
 
+type State int32
+
+/**
+  Circuit Breaker State Machine:
+
+                                 switch to open based on rule
+         +-----------------------------------------------------------------------+
+         |                                                                       |
+         |                                                                       v
++----------------+                   +----------------+      Probe      +----------------+
+|                |                   |                |<----------------|                |
+|                |   Probe succeed   |                |                 |                |
+|     Closed     |<------------------|    HalfOpen    |                 |      Open      |
+|                |                   |                |   Probe failed  |                |
+|                |                   |                +---------------->|                |
++----------------+                   +----------------+                 +----------------+
+*/
+const (
+	Closed State = iota
```

Comment:
It would be better if we could add `String()` method for it.

Reply:
ok.